### ISSUE TITLE
no-return-assign in construction like (a) => (a['b'] = 1, a)

### ISSUE THREAD

**Issue by TrueLecter** [eslint/eslint#10842]
**Tell us about your environment**

* **ESLint Version:** 5.5.0
* **Node Version:** 8.11.3
* **npm Version:** 6.4.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "extends": "standard",
    "rules": {
        "indent": ["error", 4],
        "no-unreachable": 2,
        "semi": ["error", "always"],
        "camelcase": ["error", { "properties": "never", "ignoreDestructuring": true }],
        "no-sequences": "off"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
module.exports = () => aliases.get()
    .then(ts => (_.reduce(ts, (r, t) => (r[t.alias] = t.for, r), config.aliases), ts))
    .then(ts => _(ts).groupBy('for').entries().reduce((res, [fogh, aliases]) => (res[fogh] = aliases.map(a => a.alias), res), config.reverseAliases));

```

```bash
eslint src/
```

**What did you expect to happen?** No warnings and errors

**What actually happened? Please include the actual, raw output from ESLint.** 
```
E:\Sync\Projects\NodeJS\_cq-tools\new-bot\src\preload\02-load-aliases-from-db.js
  6:31  error  Arrow function should not return assignment  no-return-assign
  7:55  error  Arrow function should not return assignment  no-return-assign
```

**Comment by ljharb** [comment]
If your arrow function contains an assignment, it should be a block form (with braces). Assignments should be treated like statements, and using the comma operator in order to use the inline arrow function form is violating this rule and also making your code less clear, for minimal value.

**Comment by ilyavolodin** [comment]
I agree. It seems like the rule is doing what it's supposed to in this case.
